### DISCUSSION TITLE
SED connection retry mechanism

### DISCUSSION THREAD

**Discussion by fgervais** [openthread/openthread#10570] [Q&A] [answered]
Hi,

I'm trying to understand what happens when a SED fails to connect to a network. Maybe it is out of range for a while. What is the retry mechanism?

All I know for now is that I call `openthread_start(ot_context)` and that's it.

Is the retry process on the openthread side or on the OS side (which would be Zephyr in my case)? Or do I, the end user, need to manage the connection retry?

**Comment by jwhui** [comment, marked as answer]
OpenThread handles the reattach behavior.

See documentation for [`OPENTHREAD_CONFIG_MLE_ATTACH_BACKOFF_ENABLE `](https://openthread.io/reference/config/group/config-mle#group__config-mle_1gaa854ba3e751b2d9281ba4fb46cacc929) for more details.

**Comment by ldoering** [comment]
Is there any api to force this reattach process to restart now? We have configured a large Backoff time, but would like to either reset the timer or try a single time on user interaction. This way we do not waste battery when the device is not in use while detached, but can respond quickly when the user uses the device.

**Reply by jwhui** [reply]
You can simply disable then enable Thread using [`otThreadSetEnabled()`](https://openthread.io/reference/group/api-thread-general#otthreadsetenabled)
